Fix for the duplicate store

princiya opened the pull request:
Issue: #54 

As specified in the issue, we now have a `storeChild.js` which is the entry point for the `lightbeam.js` to talk to the background script `store.js`. Having this setup ensures there is always one `store` instance.

princiya commented on manifest.json:
```diff
@@ -27,7 +27,6 @@
     "scripts": [
       "js/capture.js",
       "js/store.js",
-      "js/viz.js",
```
We don't need 'viz.js` to be a background script.

jonathanKingston replied:
+1

princiya commented:
I shall create a follow up PR to include the `[proxy](https://rosettacode.org/wiki/Respond_to_an_unknown_method_call#JavaScript)` object to avoid having same functions in both parent-child store objects.

Also, I plan to include all the other methods into `publicMethods`. This PR has only `getAll()`.

jonathanKingston commented on js/store.js:
```diff
@@ -93,5 +108,7 @@ function clone(obj) {
 function isObjectEmpty(obj) {
   return Object.keys(obj).length === 0;
 }
+// @todo end
```
Dunno what this is for

jonathanKingston commented on js/store.js:
```diff
@@ -93,5 +108,7 @@ function clone(obj) {
+// @todo end
 
 store.init();
+browser.runtime.onMessage.addListener(store.messageHandler.bind(store));
```
Please use arrows instead:
```
browser.runtime.onMessage.addListener(() => {
  store.messageHandler()
});
```

princiya commented:
@jonathanKingston used arrow functions.

removed unwanted code - `isEmptyObject()`. I cannot remember now why I had introduced this. I have done some code cleanup in this PR. Thanks for pointing it ou.

jonathanKingston commented:
@princiya I was actually asking about the `@todo` comment.

jonathanKingston commented:
Merging this for now. However `@todo` should be removed unless there is a reason for it.
